src/mkfs.fat.c

```diff
@@ -1372,8 +1373,22 @@ int main(int argc, char **argv)
 	    program_name = p + 1;
     }
 
-    if (gettimeofday(&create_timeval, NULL) == 0 && create_timeval.tv_sec != (time_t)-1)
+    source_date_epoch = getenv("SOURCE_DATE_EPOCH");
+    if (source_date_epoch) {
+        char *endptr = NULL;
+        long long val = 0;
+        errno = 0;
+        val = strtoll(source_date_epoch, &endptr, 10);
```

Comment:
Use `conversion` and `tmp` variables like other `strtoll` usage in this file.

Reply:
Fixed.